src/coprocessor/endpoint.rs

```diff
@@ -328,10 +333,18 @@ impl RequestTracker {
                 .with_label_values(&[self.scan_tag])
                 .observe(self.wait_time.unwrap());
         }
+
+        self.perf_statistics_start = Some(PerfStatisticsInstant::new());
     }
 
+    /// Must pair with `on_handle_start` previously.
     #[allow(useless_let_if_seq)]
-    fn record_handle(&mut self, resp: Option<&mut Response>, mut exec_metrics: ExecutorMetrics) {
+    fn on_handle_finish(&mut self, resp: Option<&mut Response>, mut exec_metrics: ExecutorMetrics) {
+        // Record delta perf statistics
+        if let Some(perf_stats) = self.perf_statistics_start {
```

Comment:
Should it be taken?

Reply:
Thanks. `take` seems to be more precise and can resist more mistakes.